enterprise/app/filter/filter.tsx

```diff
@@ -99,15 +130,50 @@ export default class FilterComponent extends React.Component<FilterProps, State>
     const startDateParam = this.props.search.get(START_DATE_PARAM_NAME);
     const endDateParam = this.props.search.get(END_DATE_PARAM_NAME);
 
-    const startDate = (startDateParam ? moment(startDateParam) : moment().subtract(7, "days")).toDate();
-    const endDate = (endDateParam ? moment(endDateParam) : moment()).toDate();
+    const defaultStartDate = getDefaultStartDateForPage(this.props.path);
+    const startDate = startDateParam ? moment(startDateParam).toDate() : defaultStartDate;
+    const endDate = endDateParam ? moment(endDateParam).toDate() : new Date();
 
     const roleValue = this.props.search.get(ROLE_PARAM_NAME) || "";
     const statusValue = this.props.search.get(STATUS_PARAM_NAME) || "";
 
     const isFiltering = Boolean(roleValue || statusValue);
     const selectedStatuses = new Set(parseStatusParam(statusValue));
 
+    const now = new Date();
+    const presetDateRanges: PresetRange[] = [
+      {
+        label: `Default (${formatDateRange(defaultStartDate, now, { now })})`,
+        isSelected: () => {
+          return !this.props.search.get(START_DATE_PARAM_NAME) && !this.props.search.get(END_DATE_PARAM_NAME);
+        },
+        range: () => ({
+          isDefault: true,
+          startDate: defaultStartDate,
+          endDate: now,
+        }),
+      },
+      ...LAST_N_DAYS_OPTIONS.map((n) => {
+        const start = moment(now)
+          .add(-n + 1, "days")
+          .toDate();
+        return {
+          label: formatDateRange(start, now, { now }),
+          isSelected: (range: Range) => {
+            return (
+              this.props.search.get(START_DATE_PARAM_NAME) &&
```

Comment:
What happens if a users selects Last 90 days and then refreshes the next day? Will they see yesterday's data and their selection is lost?